docs/standard/microservices-architecture/multi-container-microservice-net-applications/implementing-api-gateways-with-ocelot.md

```diff
@@ -0,0 +1,566 @@
+ 
+  mobilemarketingapigw:
+    image: eshop/ocelotapigw:${TAG:-latest}
+    build:
+      context: .
+      dockerfile: src/ApiGateways/ApiGw-Base/Dockerfile
+ 
+  webshoppingapigw:
+    image: eshop/ocelotapigw:${TAG:-latest}
+    build:
+      context: .
+      dockerfile: src/ApiGateways/ApiGw-Base/Dockerfile
+
+  webmarketingapigw:
+    image: eshop/ocelotapigw:${TAG:-latest}
+    build:
+      context: .
+      dockerfile: src/ApiGateways/ApiGw-Base/Dockerfile
+```
+
+Additionally, and as you can see in the docker-compose.override.yml file, the only difference between those API Gateway containers is the Ocelot configuration file which is different for each service container and specified at runtime through a Docker volume, as shown in the following docker-compose.override.yml file.
+
+```csharp
+mobileshoppingapigw:
+  environment:
+    - ASPNETCORE_ENVIRONMENT=Development
+    - IdentityUrl=http://identity.api              
+  ports:
+    - "5200:80"   
+  volumes:
+    - ./src/ApiGateways/Mobile.Bff.Shopping/apigw:/app/configuration
+ 
+mobilemarketingapigw:
+  environment:
+    - ASPNETCORE_ENVIRONMENT=Development
+    - IdentityUrl=http://identity.api              
+  ports:
+    - "5201:80"   
+  volumes:
+    - ./src/ApiGateways/Mobile.Bff.Marketing/apigw:/app/configuration
+
+webshoppingapigw:
+  environment:
+    - ASPNETCORE_ENVIRONMENT=Development
+    - IdentityUrl=http://identity.api              
+  ports:
+    - "5202:80"   
+  volumes:
+    - ./src/ApiGateways/Web.Bff.Shopping/apigw:/app/configuration
+
+webmarketingapigw:
+  environment:
+    - ASPNETCORE_ENVIRONMENT=Development
+    - IdentityUrl=http://identity.api              
+  ports:
+    - "5203:80"   
+  volumes:
+    - ./src/ApiGateways/Web.Bff.Marketing/apigw:/app/configuration
+```
+
+That is why in the source code and shown in the VS Explorer view screenshot below, the only file needed to define each specific business/BFF API Gateway is just a configuration.json file, because the four API Gateways are based on the same Docker image.
+
+![](./media/image34.png)
+
+**Figure 8-33.** The only file needed to define each API Gateway / BFF with Ocelot is a configuration file 
+
+By splitting the API Gateway into multiple API Gateways, different development teams focusing on different subsets of microservices can manage their own API Gateways by using independent Ocelot configuration files while re-using the same Ocelot Docker image. 
+
+Now, if you run eShopOnContainers with the API Gateways (included by default in VS when opening eShopOnContainers-ServicesAndWebApps.sln solution or if running “docker-compose up”), the following sample routes will be performed. 
+
+For instance, when visiting the upstream URL http://localhost:5202/api/v1/c/catalog/items/2/ served by the webshoppingapigw API Gateway, you get the result from the internal Downstream URL http://catalog.api/api/v1/2 within the Docker host, as in the following browser.
```

Comment:
bold the URLs (so they don't get translated)

A number of occurrences in this section.